### PR TITLE
Event rules

### PR DESCRIPTION
**Pull request by rait-mediarithmics**
In this PR, i created a directive eventrules to handle only event rules, i created datamart setting and i refactored sites setting to use the directive eventrules.

### CODE REVIEW

**Comment by rait-mediarithmics** [comment]
@depends on https://github.com/MEDIARITHMICS/core-platform-services/pull/145

File: app/src/core/settings/datamarts/EditOneController.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+      // ---------------- DATAMART ----------------
+
+      $scope.cancel = function() {
+        $location.path("/" + organisationId + "/settings/datamarts");
+      };
+
+      $scope.done = function() {
+        if ($scope.editMode) {
+          if ($scope.datamartToken !== $scope.datamart.token) {
+            WarningService.showWarningModal("A site token is already set. Are you sure that you want to override it?").then(sendSiteEdit, function() {
```
Comment (dduponchel) [inline app/src/core/settings/datamarts/EditOneController.js]: `A datamart token`, not `A site token`.

File: app/src/core/settings/datamarts/EditOneController.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+            error: {
+              message: "This datamart token is already taken."
+            }
+          });
+        } else {
+          ErrorService.showErrorModal(e);
+        }
+      }
+
+      function sendSiteEdit() {
```
Comment (dduponchel) [inline app/src/core/settings/datamarts/EditOneController.js]: "site" edit ?

File: app/src/core/settings/datamarts/EditOneController.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+          Restangular.all("datamarts/" + datamartId).post($scope.datamart).then(function(datamart) {
+            $scope.rules.forEach(function(ruleInfo) {
+              var rule = {
+                organisation_id: organisationId,
+                properties: ruleInfo
+              };
+              Restangular.all("datamarts/" + datamartId + "/event_rules").post(rule);
+            });
+
+            $location.path("/" + organisationId + "/settings/datamarts");
```
Comment (dduponchel) [inline app/src/core/settings/datamarts/EditOneController.js]: You should wait for the POST on event_rules to finish before redirecting (like on sites).

File: app/src/core/settings/datamarts/ViewAllController.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+define(['./module'], function (module) {
+  'use strict';
+
+  module.controller('core/settings/datamarts/ViewAllController', [
+    '$scope', '$log', '$location', '$state', '$stateParams', 'Restangular', 'core/common/auth/Session', 'lodash',
+    function ($scope, $log, $location, $state, $stateParams, Restangular, Session, _) {
+      $scope.datamartId = Session.getCurrentDatamartId();
+      $scope.organisationId = Session.getCurrentWorkspace().organisation_id;
+      $scope.itemsPerPage = 20;
+      $scope.currentPageCreative = 0;
```
Comment (dduponchel) [inline app/src/core/settings/datamarts/ViewAllController.js]: creative ?

File: app/src/core/settings/datamarts/edit.one.html [hunk 5]
```diff
@@ -0,0 +1,32 @@
+<div class="mcs-default-container" ng-controller="core/settings/datamarts/EditOneController">
+
+  <div class="mcs-default-content site-content">
```
Comment (thomas-tosoni) [inline app/src/core/settings/datamarts/edit.one.html]: datamart-content or make reusable generic css for site/datamart

**Pull request merged**